src/webgpu/api/validation/encoding/beginComputePass.spec.ts

```diff
@@ -47,11 +60,70 @@ g.test('timestampWrites,same_location')
       timestampWrites: [timestampWriteA, timestampWriteB],
     };
 
-    const encoder = t.device.createCommandEncoder();
-    const computePass = encoder.beginComputePass(descriptor);
-    computePass.end();
+    t.tryComputePass(isValid, descriptor);
+  });
 
-    t.expectValidationError(() => {
-      encoder.finish();
-    }, !isValid);
+g.test('timestampWrites,query_set_type')
+  .desc(
+    `
+  Test that all entries of the timestampWrites must have type 'timestamp'. If all query types are
+  not 'timestamp' in GPUComputePassDescriptor, a validation error should be generated.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('queryTypeA', kQueryTypes)
+      .combine('queryTypeB', kQueryTypes)
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase(['timestamp-query']);
```

Comment:
µNit: use this:
```suggestion
    t.selectDeviceForQueryTypeOrSkipTestCase(['timestamp', t.params.queryTypeA, t.params.queryTypeB]);
```

This way the test will still work in the future if we add more optional query types to kQueryTypes.

Reply:
Done!